### PR TITLE
1.8 Compound conditions Task 2

### PR DESCRIPTION
**Pull request by AlexandraPopovaz**

### CODE REVIEW

File: my-tests/setup-function-compound-conditions-task2.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/* Task 2. Write a function that defines the string submitted to the input.
+Input parameters: string
+Output parameters: string, possible values
+"numeric" - if there are only numbers
+"alpha" in the input string - if there is only Latin
+"alpha-numeric" in the input string - if both Latin and numbers
+If the string contains other characters, then return the value "unknown characters"*/
+
+    
+//1.1 option with For loop - will try to end it later
```
Comment (viraxslot) [inline my-tests/setup-function-compound-conditions-task2.ts:10]: Do you need this version?

File: my-tests/setup-function-compound-conditions-task2.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+//     //     result = 'unknown characters';
+//     // };
+//     return result;
+// };
+// console.log(isNumber(testString));
+// //export { isNumber };
+   
+    
+    
+let testString = 'fff';
```
Comment (viraxslot) [inline my-tests/setup-function-compound-conditions-task2.ts:45]: Please remove test data

File: my-tests/setup-function-compound-conditions-task2.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+function isNumber(testString: string): string | null {
```
Comment (viraxslot) [inline my-tests/setup-function-compound-conditions-task2.ts:46]: Please change the function name, it should describe the logic and it's not correct at the moment

File: my-tests/setup-function-compound-conditions-task2.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    if (!testString) {
+        return null;
+    };
+    if (typeof testString == 'number' || typeof testString == 'boolean' || typeof testString == 'object') {
```
Comment (viraxslot) [inline my-tests/setup-function-compound-conditions-task2.ts:50]: You can use ! operator here. If it's NOT a string then it's the other type.

File: my-tests/setup-function-compound-conditions-task2.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    };  
+    if  (testString.match(/[a-z]+/i) && testString.match(/[0-9]+/i)) {
+        result = 'alpha-numeric';
+    };    
+    if (result === '') {
+        return 'unknown characters';
+    } 
+    return result;
+};
+console.log(isNumber(testString));
```
Comment (viraxslot) [inline my-tests/setup-function-compound-conditions-task2.ts:68]: Please remove console.log

File: test-jest/test-compound-conditions-task2.spec.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
+import { isNumber } from "../my-tests/setup-function-compound-conditions-task2";
+
+//The first option
```
Comment (viraxslot) [inline test-jest/test-compound-conditions-task2.spec.ts:3]: Please remove the comments (and please remember about this rule in future pull requests).

File: test-jest/test-compound-conditions-task2.spec.ts [hunk 7]
```diff
@@ -0,0 +1,116 @@
+import { isNumber } from "../my-tests/setup-function-compound-conditions-task2";
```
Comment (viraxslot) [inline test-jest/test-compound-conditions-task2.spec.ts:1]: Filename is too long, I'd make it shorter

File: test-jest/test-compound-conditions-task2.spec.ts [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+//     it('Should return null if type of input parameter is an Array (object)', () => {
+//         const testData: any = [1, 2, 3];
+//         const resultFunction = isNumber(testData as string);
+//         expect(resultFunction).toBeNull();
+//     });
+// });
+
+
+//The second option - parameterized tests
+describe('Checks for isNumber function', () => {
```
Comment (viraxslot) [inline test-jest/test-compound-conditions-task2.spec.ts:62]: Don't forget to change function name here

File: test-jest/test-compound-conditions-task2.spec.ts [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            expected: null
+        },
+        {
+            title: 'Should return null if type of input parameter is an Array (object)',
+            data: [1, 2, 3],
+            expected: null
+        },
+    ]
+    testData.forEach(test => {
+        it(`should pass ${test.title}`, () => {
```
Comment (viraxslot) [inline test-jest/test-compound-conditions-task2.spec.ts:111]: Finally, the name will be "should pass Should return" and it doesn't look good. I've added "should pass" for the example.

File: test-jest/test-compound-conditions-task2.spec.ts [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+//         const testData: any = [1, 2, 3];
+//         const resultFunction = isNumber(testData as string);
+//         expect(resultFunction).toBeNull();
+//     });
+// });
+
+
+//The second option - parameterized tests
+describe('Checks for isNumber function', () => {
+    const testData = [
```
Comment (viraxslot) [inline test-jest/test-compound-conditions-task2.spec.ts:63]: What about empty string, 1-symbol string, numbers/special characters in the beginning/end?

File: test-jest/test-compound-conditions.spec.ts [hunk 11]
```diff
@@ -1,137 +1,126 @@
...
+        },
+        {
+            title: 'should return the second array if sums of values and amount of odd numbers are equal',
+            testArray1: [1, 2],
+            testArray2: [2, 1],
+            expected: [2, 1]
+        },
+    ]
+    testData.forEach(test => {
+        it(`should pass ${test.title}`, () => {
```
Comment (viraxslot) [inline test-jest/test-compound-conditions.spec.ts:121]: I believe you can write it like `it(test.title, () => {})`

File: my-tests/setup-function-compound-conditions-task2.ts (hunk 1 above)
Comment (AlexandraPopovaz) [reply my-tests/setup-function-compound-conditions-task2.ts]: Removed

File: my-tests/setup-function-compound-conditions-task2.ts (hunk 2 above)
Comment (AlexandraPopovaz) [reply my-tests/setup-function-compound-conditions-task2.ts]: Removed

File: my-tests/setup-function-compound-conditions-task2.ts (hunk 3 above)
Comment (AlexandraPopovaz) [reply my-tests/setup-function-compound-conditions-task2.ts]: Thanks, changed

File: my-tests/setup-function-compound-conditions-task2.ts (hunk 4 above)
Comment (AlexandraPopovaz) [reply my-tests/setup-function-compound-conditions-task2.ts]: Thanks, fixed it

File: my-tests/setup-function-compound-conditions-task2.ts (hunk 5 above)
Comment (AlexandraPopovaz) [reply my-tests/setup-function-compound-conditions-task2.ts]: Removed

File: test-jest/test-compound-conditions-task2.spec.ts (hunk 7 above)
Comment (AlexandraPopovaz) [reply test-jest/test-compound-conditions-task2.spec.ts]: Fixed

File: test-jest/test-compound-conditions-task2.spec.ts (hunk 8 above)
Comment (AlexandraPopovaz) [reply test-jest/test-compound-conditions-task2.spec.ts]: Ok

File: test-jest/test-compound-conditions-task2.spec.ts (hunk 9 above)
Comment (AlexandraPopovaz) [reply test-jest/test-compound-conditions-task2.spec.ts]: Fixed

File: test-jest/test-compound-conditions-task2.spec.ts (hunk 10 above)
Comment (AlexandraPopovaz) [reply test-jest/test-compound-conditions-task2.spec.ts:4]: Thanks! Added some checks

File: my-tests/compound-conditions-func2.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+        return null;
+    };
+    if (typeof testString !== 'string') {
+        return null;
+    };
+    let result: string = '';  
+    if (testString.match(/^([a-z][A-Z]+)$/i)) {
+        result = 'alpha';
+    };   
+    if (testString.length === 1) {
```
[the remark this answers is not in the document]
Comment (AlexandraPopovaz) [reply my-tests/compound-conditions-func2.ts]: Thanks! Fixed

**Review by viraxslot — APPROVED** [review]

**Pull request merged**